### PR TITLE
[Impeller] partially remove remap sampler support

### PR DESCRIPTION
**Pull request by jonahwilliams**
Part of https://github.com/flutter/flutter/issues/119173. The runtime IPLR contains all of the infromation to handle samplers without remaping. Its just a bit awkward to use at this point.

As a follow up will make location data take into account this offset and delete the flag (after rolling into tool).

Separately, we should investigate if enabling vulkan regressed any other functionality on macOS.

### CODE REVIEW

**Comment by jonahwilliams** [comment]
I could also fixup the location data in the generated IPLR file but I'm unusure if we're using it for anything else.

File: impeller/entity/contents/runtime_effect_contents.cc [hunk 1]
```diff
@@ -210,6 +207,36 @@ bool RuntimeEffectContents::Render(const ContentContext& renderer,
     }
   }
 
+  size_t sampler_index = 0;
+  FML_DCHECK(minimum_sampler_index >= 0);
```
[the remark this answers is not in the document]
Comment (jonahwilliams) [reply impeller/entity/contents/runtime_effect_contents.cc:211]: RIGHT

[the remark this answers is not in the document]
Comment (jonahwilliams) [reply impeller/entity/contents/runtime_effect_contents.cc]: fixed

**Review by iskakaushik — APPROVED** [review]
LGTM

File: testing/dart/fragment_shader_test.dart [hunk 2]
```diff
@@ -332,34 +332,6 @@ void main() async {
     shader.dispose();
   });
 
-  // This test can't rely on actual pixels rendered since it needs to run on a
```
Comment (zanderso) [inline testing/dart/fragment_shader_test.dart:335]: Is the test on line 275 sufficient? Does that test need to have a sampler or two thrown in?

Comment (jonahwilliams) [reply testing/dart/fragment_shader_test.dart:335]: wouldn't apply either way, its not running on impeller or metal?

**Pull request merged**